src/Microsoft.CodeQuality.Analyzers/UnitTests/Performance/DoNotUseCountWhenAnyCanBeUsedTests.cs

```diff
@@ -0,0 +1,550 @@
+        [MemberData(nameof(DoNotUseCountWhenAnyCanBeUsedTestData.RightCount_NoDiagnostic_TheoryData), MemberType = typeof(DoNotUseCountWhenAnyCanBeUsedTestData))]
+        public void Basic_NoDiagnostic_RightCount(int value, BinaryOperatorKind @operator)
+        {
+            VerifyBasic(
+                $@"
+Imports System
+Imports System.Linq
+Class C
+    Sub S()
+        Dim b = {value} {BasicOperatorText(@operator)} Enumerable.Range(0, 0).Count()
+    End Sub
+End Class
+");
+        }
+
+        [Fact]
+        public void CSharp_NoDiagnostic_NotEnumerableCountEqualsConstant()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Linq;
+namespace N
+{
+    class C
+    {
+        void M()
+        {
+            var b = Enumerable.Range(0, 0).Count() == 0;
+        }
+    }
+    static class E
+    {
+        public static int Count<TSource>(this System.Collections.Generic.IEnumerable<TSource> source) => 0;
+    }
+}");
+        }
+
+        [Fact]
+        public void Basic_NoDiagnostic_NotEnumerableCountEqualsConstant()
+        {
+            VerifyBasic(
+                $@"
+Imports System
+Imports System.Linq
+Namespace N
+    Class C
+        Sub S()
+            Dim b = Enumerable.Range(0, 0).Count() = 0
+        End Sub
+    End Class
+    Module M
+        <System.Runtime.CompilerServices.ExtensionAttribute()>
+        Public Function Count(Of TSource)(ByVal source As System.Collections.Generic.IEnumerable(Of TSource)) As Integer
+            Return 0
+        End Function
+    End Module
+End Namespace
+");
+        }
+
+        [Fact]
+        public void CSharp_NoDiagnostic_EqualsNotEnumerableCount()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Linq;
+namespace N
+{
+    class C
+    {
+        void M()
+        {
+            var b = 0 == Enumerable.Range(0, 0).Count();
+        }
+    }
+    static class E
+    {
+        public static int Count<TSource>(this System.Collections.Generic.IEnumerable<TSource> source) => 0;
+    }
+}");
+        }
+
+        [Fact]
+        public void Basic_NoDiagnostic_EqualsNotEnumerableCount()
+        {
+            VerifyBasic(
+                $@"
+Imports System
+Imports System.Linq
+Namespace N
+    Class C
+        Sub S()
+            Dim b = 0 = Enumerable.Range(0, 0).Count()
+        End Sub
+    End Class
+    Module M
+        <System.Runtime.CompilerServices.ExtensionAttribute()>
+        Public Function Count(Of TSource)(ByVal source As System.Collections.Generic.IEnumerable(Of TSource)) As Integer
+            Return 0
+        End Function
+    End Module
+End Namespace
+");
+        }
+
+        #endregion
+
+        #region Unit tests for analyzer diagnostic(s)
+
+        [Theory]
+        [MemberData(nameof(DoNotUseCountWhenAnyCanBeUsedTestData.LeftCount_Diagnostic_TheoryData), MemberType = typeof(DoNotUseCountWhenAnyCanBeUsedTestData))]
+        public void CSharp_Diagnostic_LeftCount(BinaryOperatorKind @operator, int value)
```

Comment:
💡  _All_ diagnostic tests should use `VerifyCodeFixAsync`. Separating the code fix tests serves only to increase complexity and increase the probability of a test gap. I'm slowly working to update the old tests in this repository to use the recommended pattern, but new tests for analyzers that have a fixer should start that way.

Reply:
@sharwell, do you mean that, in this particular case, I should remove these tests and keep only the fixer tests. They are the same use cases.

Reply:
@paulomorgado If you have a diagnostic test and a code fix test that have the same input source code, then you can simply remove the diagnostic test and keep the existing code fix test. `VerifyCodeFixAsync` automatically verifies the diagnostics of both the source and fixed code, so the code fix in a case like this would be exactly duplicating work done in the diagnostic test.

However, it's common for users getting started with the new test framework to find that one or more diagnostic tests do not already have a code fix test present with the same input. In those cases, it would be good to update the diagnostic test rather than simply removing it.